operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/NamedEventSource.java

```diff
@@ -1,19 +1,27 @@
 package io.javaoperatorsdk.operator.processing.event;
 
 import java.util.Objects;
+import java.util.Optional;
 
 import io.javaoperatorsdk.operator.OperatorException;
 import io.javaoperatorsdk.operator.processing.event.source.EventSource;
 import io.javaoperatorsdk.operator.processing.event.source.EventSourceStartPriority;
 
-class NamedEventSource implements EventSource {
+class NamedEventSource implements EventSource, EventSourceMetadata {
 
   private final EventSource original;
   private final String name;
+  private final AssociatedDependentMetadata dependentMetadata;
```

Comment:
Porbably not fully understand the PR, but really don't like this part. The event sources should have no knowledge at all about dependent resources.